### ISSUE TITLE
Multi-segment class imports are not correctly identified

### ISSUE THREAD

**Issue by IGJoshua** [clj-kondo/clj-kondo#1283]
**version**

v2021.04.23

**platform**

Native, installed from AUR

**editor**

Emacs, flycheck-clj-kondo

**problem**

Imports which include a part of the package aren't recognized correctly.

**repro**

```clojure
(ns repro
  (:import
    (java.time format.DateTimeFormatter LocalDateTime)))

(defn parse-date
  [s]
  (LocalDateTime/parse s DateTimeFormatter/ISO_DATE_TIME))
```

In the above file, `format.DateTimeFormatter` is reported as an unused import, and `DateTimeFormatter/ISO_DATE_TIME` is reported as an unresolved namespace.

```
$ clj-kondo --lint repro.clj
repro.clj:3:16: warning: Unused import format.DateTimeFormatter
repro.clj:7:26: warning: Unresolved namespace DateTimeFormatter. Are you missing a require?
linting took 11ms, errors: 0, warnings: 2
```

**config**

The bug occurs without configuration.

**expected behavior**

The class name with a package component to be recognized as imported.

**Comment by borkdude** [comment]
@IGJoshua I discussed this on Clojurians Slack and it seems that it's an accident that this is supported. According to the docstring you should use `(package-name classes*)` and not package segments as part of the class name.

<img width="749" alt="Screenshot 2021-05-21 at 18 29 42" src="https://user-images.githubusercontent.com/284934/119169732-a7d13880-ba62-11eb-822a-41119943ddba.png">

**Comment by IGJoshua** [comment]
Alright, well thanks for looking into this! Maybe you could add a lint to discourage this usage?

**Comment by borkdude** [comment]
Addendum 

<img width="788" alt="Screenshot 2021-05-21 at 19 06 20" src="https://user-images.githubusercontent.com/284934/119173516-a8200280-ba67-11eb-94a0-64e27c08607e.png">

**Comment by IGJoshua** [comment]
Yes, kondo does already have findings for these cases, like the ones shown in the original comment for this issue, however the actual message it's reported with does not match the issue that is occurring. The case that I've shown here says that a namespaces is unresolved, and one import is unused. Both of those things are untrue, but it is true that using a multi-segment class import is undefined behavior. This leads me to believe that a new lint should be used to report this issue, rather than leaving it to the current (and incorrect) messages.

**Comment by borkdude** [comment]
@IGJoshua I wasn't responding to your latest reply, just wanted to add some information to this thread. You're welcome to improve the error messages reported by clj-kondo.
